overlord/hookstate/hooktest/handler.go

```diff
@@ -41,6 +45,10 @@ func NewMockHandler() *MockHandler {
 
 // Before satisfies hookstate.Handler.Before
 func (h *MockHandler) Before() error {
+	executing := atomic.AddInt32(&h.Executing, 1)
+	if executing != 1 {
+		panic(fmt.Sprintf("More than one handler executed: %d", executing))
```

Comment:
Again, can we please not do that in this type? This is hardcoding an assumption in a publicly exposed test type that is simply not a real constraint and can blow things up without the logic in the test being wrong.